pkg/controller/tsp/status.go

```diff
@@ -0,0 +1,382 @@
+package tsp
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gocrane/api/prediction/v1alpha1"
+
+	"github.com/gocrane/crane/pkg/common"
+	"github.com/gocrane/crane/pkg/known"
+	"github.com/gocrane/crane/pkg/prediction"
+)
+
+// Scan all TimeSeriesPredictions and update the status if it is needed,
+// update each time series prediction status window length is double of the spec.PredictionWindowSeconds.
+// check the actual state of world and decide if need to update the crd status, it is periodic check to meet updateStatusDelayQueue's flaw.
+func (tc *Controller) syncPredictionsStatus() error {
```

Comment:
what is the difference between syncPredictionsStatus and updateStatusDelayQueue? they all update tsp status

Reply:
updateStatusDelayQueue must do predict when the time is ready based on the PredictionWindowSeconds in crd. when it update, next PredictionWindowSeconds it will enqueue again.

But this is not enough, when user change the PredictionWindowSeconds or change the algorithm, we should update it, periodic check it better. we can remove updateStatusDelayQueue if we just use periodic check.